### PR TITLE
fix(zendesk): only fetch comment authors

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

We were trying to fetch all users in the workspace every single time, which was taking a very long time (never really completing from what I can tell)

With this change, we're only fetching the users that commented on the tickets, which is usually very quick.

## Risk

N/A

## Deploy Plan

Deploy connectors and resume the connectors

### CODE REVIEW

**Review by aubin-tchoi — APPROVED** [review]

**Comment by aubin-tchoi** [comment]
LGTM 🙏

File: connectors/src/connectors/zendesk/temporal/activities.ts [hunk 1]
```diff
@@ -434,16 +434,18 @@ export async function syncZendeskTicketBatchActivity({
...
-  const comments = await concurrentExecutor(
+  const comments2d = await concurrentExecutor(
     tickets,
     async (ticket) => zendeskApiClient.tickets.getComments(ticket.id),
     { concurrency: 3, onBatchComplete: heartbeat }
   );
+  const userIds = _.uniq(
+    _.flatten(comments2d.map((comments) => comments.map((c) => c.author_id)))
+  );
+  const { result: users } = await zendeskApiClient.users.showMany(userIds);
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: Whic part :p ?

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: i so deeply hate Lodash it's crazy

[the remark this answers is not in the document]
Comment (fontanierh) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: Why ? it's pretty convenient.

WIthout lodash you'd need to do:

```
Array.from(new Set(comments2d.map((comments) => comments.map((c) => c.author_id)).flat()))
```

which in my opinion is less readable: 
- array.from + set is longer
- mixing fn(...) with obj.meth(...) syntaxes on one line makes it harder to understand

debatable ofc, but we have lodash so we might as well use it

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: I would personally do `const userIds = [...new Set(comments2d.flatMap(comments => comments.map(c => c.author_id)))];`

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: true lol

[the remark this answers is not in the document]
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/activities.ts:445]: thank god we do rust

**Pull request merged**